extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -243,17 +231,12 @@ public void run() {
     private void stopJobs() {
         logger.debug("Stopping jobs for thing {}", getThing().getUID());
         synchronized (schedulerLock) {
-            if (quartzScheduler != null) {
-                try {
-                    String thingUid = getThing().getUID().toString();
-                    for (JobKey jobKey : quartzScheduler.getJobKeys(jobGroupEquals(thingUid))) {
-                        logger.debug("Deleting astro {} for thing '{}'", jobKey.getName(), thingUid);
-                        quartzScheduler.deleteJob(jobKey);
-                    }
-                } catch (SchedulerException ex) {
-                    logger.error("{}", ex.getMessage(), ex);
-                }
-            }
+            // String thingUid = getThing().getUID().toString();
```

Comment:
Without having an IDE open, does the scheduler return a view of the queue or a copy?
I assume the queue could be changed by the scheduler while iteration.